### PR TITLE
Set win lose

### PR DESCRIPTION
**Pull request by Heidi9287**
in this branch I did below
1. changed the setup game button to a timed popup and added a close button and some css to it
2. wrote winner/loser component in tsx to show the corresponding component when the "gameStatus" === complete
3. added basic css to it

### CODE REVIEW

File: src/components/RoomView/Loser.tsx
Comment (Joshowens16) [inline src/components/RoomView/Loser.tsx:26]: We should really avoid using "any" in typescript, could you make the type more specific?

File: src/components/RoomView/Winner.tsx
Comment (Joshowens16) [inline src/components/RoomView/Winner.tsx:31]: Same here, we should something other than any

File: src/components/RoomView/Winner.tsx
Comment (Joshowens16) [inline src/components/RoomView/Winner.tsx:17]: This needs to use RootState, not ReduxState. The RootState is just export from the Redux store so need to define it here

Comment (Joshowens16) [reply src/components/RoomView/Winner.tsx:17]: Heres an example:
Screen Shot 2023-02-04 at 2 13 30 PM

**Review by Joshowens16 — CHANGES_REQUESTED** [review]
I'm not really sure this can be approved, it doesn't use the correct version of redux state and we might need to merge the frontendTS branch before we can merge this

File: src/components/RoomView/Loser.tsx
Comment (Heidi9287) [reply src/components/RoomView/Loser.tsx:26]: I can't find a way to avoid using any here other than defining all the types in the interface, which feels a bit redundant for a helper function. Because we are using a "for..in" in the next couple of lines to loop through the object of team-1/team-2: {spymaster: {playerID: someId, username: name} operator:{firebaseKey{playerID: someId, username: name}}.

File: src/components/RoomView/Winner.tsx
Comment (Heidi9287) [reply src/components/RoomView/Winner.tsx:17]: we can wait for the front end ts merged to deal with this pull request

File: src/components/RoomView/roomView.css
Comment (rosalie0) [inline src/components/RoomView/roomView.css:108]: Not sure exactly where the problem lays but this displays in a weird position for me. additionally, i'm not sure if we want this to be a huge popup over the board - because as soon as you win or lose it may just cover the screen and you can't tell what happened.
eg) i clicked the black card and suddenly i just have a big message saying i lost on the screen and i have no idea what happened, only once i close that can i see the board, and see that the black card was revealed.

however this is just css and we can adjust later. i don't want to hold up the logic for this PR with just styling stuff.

File: src/components/RoomView/RoomView.tsx
Comment (rosalie0) [reply src/components/RoomView/RoomView.tsx:303]: now that i think about it - I don't think this needs to a popup at all - but instead just a component that we render if you are the host && game status is ready, and instead of the close button, when you click the start game button , it does not render.
we can discuss this in class.
maybe the best thing to do for now is to push this onto main as is, and then make a new PR to change it.

**Review by rosalie0 — CHANGES_REQUESTED** [review]

File: src/components/RoomView/RoomView.tsx
Comment (Heidi9287) [reply src/components/RoomView/RoomView.tsx:8]: we can

File: src/components/RoomView/RoomView.tsx
Comment (Heidi9287) [reply src/components/RoomView/RoomView.tsx:296]: because the outside div contains two functions: gameStatus and ClaimHost, therefore the outside div name is gameStatusClaimHost, the innder div is gameStatus, I am using gameStatus on claimHost function as well for now but open the possibility of styleing it separtely, in that case, will change the className to claimHost

Comment (rosalie0) [reply src/components/RoomView/RoomView.tsx:296]: ok! changing the names sounds good, i just wanted to make sure of the intended order!

**Review by rosalie0 — APPROVED** [review]

**Review by Joshowens16 — APPROVED** [review]

**Pull request merged**
